modules/dnn/src/ocl4dnn/src/ocl4dnn_conv_spatial.cpp

```diff
@@ -594,10 +594,10 @@ bool OCL4DNNConvSpatial<Dtype>::Forward(const UMat& bottom,
     if (use_half_ && weights_half.empty())
         convertFp16(weight, weights_half);
 
-    prepareKernel(bottom, top, weight, (use_half_) ? bias_half : bias, numImages);
+    prepareKernel(bottom, top, (use_half_) ? weights_half : weight, (use_half_) ? bias_half : bias, numImages);
     if (bestKernelConfig.empty())
         return false;
-    return convolve(bottom, top, weight, (use_half_) ? bias_half : bias, numImages, bestKernelConfig);
+    return convolve(bottom, top, (use_half_) ? weights_half : weight, (use_half_) ? bias_half : bias, numImages, bestKernelConfig);
```

Comment:
Looks like OpenCL kernels are still called properly (without this patch).

There are several other places with handling this in this file: [1](https://github.com/opencv/opencv/blob/4.5.0/modules/dnn/src/ocl4dnn/src/ocl4dnn_conv_spatial.cpp#L1107-L1110) [2](https://github.com/opencv/opencv/blob/4.5.0/modules/dnn/src/ocl4dnn/src/ocl4dnn_conv_spatial.cpp#L1151-L1154) [3](https://github.com/opencv/opencv/blob/4.5.0/modules/dnn/src/ocl4dnn/src/ocl4dnn_conv_spatial.cpp#L1959)

IMHO, it is a coding issue.
There is no reason to pass FP32 weight/bias to this implementation.
FP16 should be passed instead by caller (convolution_layer.cpp).

Also it should avoid double conversion FP16->FP32->FP16 hacks like [this](https://github.com/opencv/opencv/blob/4.5.0/modules/dnn/src/layers/convolution_layer.cpp#L1589).